akka-http/src/main/scala/akka/http/scaladsl/server/HttpApp.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.scaladsl.server
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.stream.ActorMaterializer
+import akka.http.scaladsl.settings.ServerSettings
+import com.typesafe.config.{ Config, ConfigFactory }
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ Await, ExecutionContext, Future }
+import scala.io.StdIn
+import scala.util.{ Failure, Success }
+
+/**
+ * Bootstrap trait. It helps booting up an akka-http by only defining the desired routes.
+ * It offers additional hooks to modify the default behavior
+ */
+trait HttpApp {
+
+  def main(args: Array[String]) = {
+    val config = createConfig()
+    implicit val system = createActorSystem(config)
+    implicit val materializer = ActorMaterializer()
+    implicit val executionContext = system.dispatcher
+    postStartActorSystem(system)
+
+    val host = config.getString("akka.http.server.host")
+    val port = config.getInt("akka.http.server.port")
```

Comment:
Good input @lihaoyi, thanks a lot! :)

FYI:

> Also, I often like converting config into to type-safe settings objects as soon as possible rather than waiting for some library to try (and possibly fail) on my behalf.

FYI, we're working on something that'll allow more programatic and typesafe configuration: https://github.com/akka/akka/pull/21922